activestorage/storage.py

```diff
@@ -48,19 +48,8 @@ def reduce_chunk(rfile, offset, size, compression, filters, missing, dtype, shap
             # no valid data would have to do something like this too.
             result = method(tmp)
             if np.ma.is_masked(result):
-                # FIXME dodgy
-                prct = {False: 1.1, True: 0.9}
-                if missing[0]:
-                    return missing[0]
-                if missing[1]:
-                    return missing[1]
-                #FIXME how do we avoid fail with over/under flow?
-                if missing[2]:
-                    norm_factor = prct[missing[2]>0]
-                    return missing[2] * norm_factor
-                if missing[3]:
-                    norm_factor = prct[missing[3]<0]
-                    return missing[3] * norm_factor
+                nonmasked_result = result[~result.mask].data
```

Comment:
thanks @davidhassell - this was me a bit optimistic that we'll eventually return arrays instead of single floats, I'll change it to get a single floaty-float :+1:

Reply:
done! Tests pass nicely too :grin: